eduNEXT/platform-plugin-forum-email-notifier
# feat: add `es_419` and `es_ES` translations

### Description
This PR adds translation support to the plugin. At the moment it only adds the translation to "Spanish (Latin America)" and "Spanish (Spain)".

### How to Test
For this plugin to work properly you should have the following requirements:
1. Install `tutor-forum` plugin
2. Enable the plugin and run `tutor config save`
3. Build the tutor forum image: `tutor images build forum`
4. Install `openedx-events` in `v9.0.0` or higher.
5. Install `edx-platform` with this [branch](https://github.com/openedx/edx-platform/pull/33395)
6. In your environment, configure the following settings. The following plugin is used as reference:

    ```yaml
    name: settings
    version: 0.1.0
    patches:
        openedx-common-settings: |
          OPEN_EDX_FILTERS_CONFIG = {
            "org.openedx.learning.instructor.dashboard.render.started.v1": {
              "fail_silently": False,
              "pipeline": [
                "platform_plugin_forum_email_notifier.extensions.filters.AddInstructorNotifierTab",
              ]
            },
          }
          LOCALE_PATHS += ["/openedx/requirements/platform-plugin-forum-email-notifier/platform_plugin_forum_email_notifier/conf/locale"]

    ```

Afterwards:
#### Emails
1. Install this branch in your environment
2. Go to the discussions page
3. Create a new post
4. An email notification should be sent. Those can be found in the lms container in the folder: `tmp/openedx/emails`. You should see the email translated into your preferred language

#### Instructor Dashboard
1. Go to the instructor dashboard
2. Select the forum notifications tab
3. You should see all text translated into your preferred language

### Screenshots
#### Emails
- Creation of a post

    ![image](https://github.com/eduNEXT/platform-plugin-forum-email-notifier/assets/64033729/4c624925-a9ea-4f96-b0eb-e3141339cb59)
- Response of a post

    ![image](https://github.com/eduNEXT/platform-plugin-forum-email-notifier/assets/64033729/e6998642-7523-498b-8d07-ba87b3640238)
- Comment of a post

    ![image](https://github.com/eduNEXT/platform-plugin-forum-email-notifier/assets/64033729/ff47ed5a-c90d-4125-9679-a0dfc88b6f0b)

- Digest

    ![image](https://github.com/eduNEXT/platform-plugin-forum-email-notifier/assets/64033729/93daed59-59dd-43d9-8984-78715a7166fc)

#### Instructor Dashboard
![image](https://github.com/eduNEXT/platform-plugin-forum-email-notifier/assets/64033729/a5889dea-f3b9-4946-a336-ba8bbcfaee15)
